tensorflow/compiler/xla/service/mlir_gpu/tests/mlir_gpu_lhlo_gen_test.cc

```diff
@@ -393,5 +393,162 @@ ENTRY %AddReduce (x: f32[100,10], c: f32[]) -> f32[100] {
+ENTRY %Ceil (val: f32[2,2]) -> f32[2,2] {
+  %val = f32[2,2]{1,0} parameter(0)
+  ROOT %ceil = f32[2,2]{1,0} ceil(f32[2,2]{1,0} %val)
+})",
+                     R"(
+;CHECK: func @ceil(%[[ARG0:.*]]: [[TYPE:.*]], %[[ARG1:.*]]: [[TYPE]]) {
+;CHECK:   "xla_lhlo.ceil"(%[[ARG0]], %[[ARG1]]) : ([[TYPE]], [[TYPE]]) -> ()
+;CHECK: }
+      )");
+}
+
+TEST_F(LhloGenTest, Convert) {
+  CompileAndVerifyIr(R"(
+HloModule Convert
+ENTRY %Convert (val: i32[2,2]) -> f32[2,2] {
+  %val = i32[2,2]{1,0} parameter(0)
+  ROOT %convert = f32[2,2]{1,0} convert(i32[2,2]{1,0} %val)
+})",
+                     R"(
+;CHECK: func @convert(%[[ARG0:.*]]: [[TYPE:.*]], %[[ARG1:.*]]: [[TYPE]]) {
+;CHECK:   "xla_lhlo.convert"(%[[ARG0]], %[[ARG1]]) : ([[TYPE]], [[TYPE]]) -> ()
+;CHECK: }
+      )");
+  CompileAndVerifyIr(R"(
+HloModule Convert
+ENTRY %Convert (val: f32[2,2]) -> f64[2,2] {
+  %val = f32[2,2]{1,0} parameter(0)
+  ROOT %convert = f64[2,2]{1,0} convert(f32[2,2]{1,0} %val)
+})",
+                     R"(
+;CHECK: func @convert(%[[ARG0:.*]]: [[TYPE:.*]], %[[ARG1:.*]]: [[TYPE]]) {
+;CHECK:   "xla_lhlo.convert"(%[[ARG0]], %[[ARG1]]) : ([[TYPE]], [[TYPE]]) -> ()
+;CHECK: }
+      )");
+  CompileAndVerifyIr(R"(
+HloModule Convert
+ENTRY %Convert (val: f64[2,2]) -> f32[2,2] {
+  %val = f64[2,2]{1,0} parameter(0)
+  ROOT %convert = f32[2,2]{1,0} convert(f64[2,2]{1,0} %val)
+})",
+                     R"(
+;CHECK: func @convert(%[[ARG0:.*]]: [[TYPE:.*]], %[[ARG1:.*]]: [[TYPE]]) {
+;CHECK:   "xla_lhlo.convert"(%[[ARG0]], %[[ARG1]]) : ([[TYPE]], [[TYPE]]) -> ()
+;CHECK: }
+      )");
+  CompileAndVerifyIr(R"(
+HloModule Convert
+ENTRY %Convert (val: i32[2,2]) -> i8[2,2] {
+  %val = i32[2,2]{1,0} parameter(0)
+  ROOT %convert = i8[2,2]{1,0} convert(i32[2,2]{1,0} %val)
+})",
+                     R"(
+;CHECK: func @convert(%[[ARG0:.*]]: [[TYPE:.*]], %[[ARG1:.*]]: [[TYPE]]) {
+;CHECK:   "xla_lhlo.convert"(%[[ARG0]], %[[ARG1]]) : ([[TYPE]], [[TYPE]]) -> ()
+;CHECK: }
+      )");
+  CompileAndVerifyIr(R"(
+HloModule Convert
+ENTRY %Convert (val: i8[2,2]) -> i32[2,2] {
+  %val = i8[2,2]{1,0} parameter(0)
+  ROOT %convert = i32[2,2]{1,0} convert(i8[2,2]{1,0} %val)
+})",
+                     R"(
+;CHECK: func @convert(%[[ARG0:.*]]: [[TYPE:.*]], %[[ARG1:.*]]: [[TYPE]]) {
+;CHECK:   "xla_lhlo.convert"(%[[ARG0]], %[[ARG1]]) : ([[TYPE]], [[TYPE]]) -> ()
+;CHECK: }
+      )");
+}
+
+TEST_F(LhloGenTest, Cos) {
+  CompileAndVerifyIr(R"(
+HloModule Cos
+ENTRY %Cos (val: f32[2,2]) -> f32[2,2] {
+  %val = f32[2,2]{1,0} parameter(0)
+  ROOT %cos = f32[2,2]{1,0} cos(f32[2,2]{1,0} %val)
+})",
+                     R"(
+;CHECK: func @cos(%[[ARG0:.*]]: [[TYPE:.*]], %[[ARG1:.*]]: [[TYPE]]) {
+;CHECK:   "xla_lhlo.cos"(%[[ARG0]], %[[ARG1]]) : ([[TYPE]], [[TYPE]]) -> ()
+;CHECK: }
+      )");
+}
+
+TEST_F(LhloGenTest, Neg) {
+  CompileAndVerifyIr(R"(
+HloModule Neg
+ENTRY %Neg (val: f32[2,2]) -> f32[2,2] {
+  %val = f32[2,2]{1,0} parameter(0)
+  ROOT %neg = f32[2,2]{1,0} neg(f32[2,2]{1,0} %val)
```

Comment:
neg -> negate for the HLO op.
Similar to the cosine fix, you will also have to adjust func below.